mirrord-layer/src/file.rs

```diff
@@ -106,3 +117,299 @@ impl OpenOptionsInternalExt for OpenOptionsInternal {
             })
     }
 }
+
+type ResponseDeque<T> = VecDeque<ResponseChannel<T>>;
```

Comment:
Kind of an extreme type idea, but maybe adding a marker trait here to guarantee that this cannot be used with any type would be nice?

Illustration:

```rust
trait FileOperation {}

type ResponseDeque<T: FileOperation> = ...;

impl FileOperation for OpenFileResponse {}

...
```

This would prevent possible (but very unlikely) misuses of the type.

Reply:
I think that would be an overkill and might impact compile time with no real value..